ecl/resource_ecl_baremetal_server_v2.go

```diff
@@ -0,0 +1,593 @@
+	// If an image_name was specified, look up the image ID, report if error.
+	imageID, err := getBaremetalImageID(baremetalClient, d)
+	if err != nil {
+		return err
+	}
+
+	flavorID, err := getBaremetalFlavorID(baremetalClient, d)
+	if err != nil {
+		return err
+	}
+
+	networks, err := getBaremetalNetworks(baremetalClient, d)
+	if err != nil {
+		return err
+	}
+
+	raidArrays, err := getBaremetalRaidArrays(baremetalClient, d)
+	if err != nil {
+		return err
+	}
+
+	lvmVolumeGroups, err := getBaremetalLVMVolumeGroups(baremetalClient, d)
+	if err != nil {
+		return err
+	}
+
+	filesystems, err := getBaremetalFilesystems(baremetalClient, d)
+	if err != nil {
+		return err
+	}
+
+	personality, err := getBaremetalPersonalities(baremetalClient, d)
+	if err != nil {
+		return err
+	}
+
+	createOpts = &servers.CreateOpts{
+		Name:             d.Get("name").(string),
+		AdminPass:        d.Get("admin_pass").(string),
+		KeyName:          d.Get("key_pair").(string),
+		ImageRef:         imageID,
+		FlavorRef:        flavorID,
+		AvailabilityZone: d.Get("availability_zone").(string),
+		Networks:         networks,
+		Metadata:         resourceInstanceMetadataV2(d),
+		UserData:         []byte(d.Get("user_data").(string)),
+		RaidArrays:       raidArrays,
+		LVMVolumeGroups:  lvmVolumeGroups,
+		Filesystems:      filesystems,
+		Personality:      personality,
+	}
+
+	log.Printf("[DEBUG] Create Options: %#v", createOpts)
+
+	var server *servers.Server
+	server, err = servers.Create(baremetalClient, createOpts).Extract()
+	if err != nil {
+		return fmt.Errorf("Error creating baremetal server: %s", err)
+	}
+
+	log.Printf("[INFO] baremetal server ID: %s", server.ID)
+
+	log.Printf("[DEBUG] Waiting for baremetal server (%s) to become available", server.ID)
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"BUILD"},
+		Target:     []string{"ACTIVE"},
+		Refresh:    waitForBaremetalServerActive(baremetalClient, server.ID),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      5 * time.Second,
+		MinTimeout: 30 * time.Second,
+	}
+
+	d.SetId(server.ID)
+	d.Set("admin_pass", server.AdminPass)
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf(
+			"Error waiting for baremetal server (%s) to become ready: %s",
+			server.ID, err)
+	}
+
+	log.Printf(
+		"[DEBUG] Waiting for baremetal server (%s) to become running",
+		server.ID)
+
+	return resourceBaremetalServerV2Read(d, meta)
+}
+
+func resourceBaremetalServerV2Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	baremetalClient, err := config.baremetalV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating ECL baremetal client: %s", err)
+	}
+
+	server, err := servers.Get(baremetalClient, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "server")
+	}
+
+	log.Printf("[DEBUG] Retrieved Server %s: %+v", d.Id(), server)
+
+	d.Set("name", server.Name)
```

Comment:
Why you only set name as state is, bare-metal API does not support Update. right?
But this may cause unexpected change in case updating configurations.
Under the condition that terraform only has server name as state, Terraform may not sense configuration change excepting name. (I can not say confidently)

To confirm it, could you please add ACC test to explicitly cause ForceNew change like compute server?

Reply:
Okay, let me check it.

Reply:
As we discussed yesterday, I've removed state "name" because there is no update API:)